### DISCUSSION TITLE
[RFC] The use of `pnpm`

### DISCUSSION THREAD

**Discussion by dcyoung-dev** [DCC-EX/EX-WebThrottle2#29] [Ideas]
### Problem

Onboarding new developers can be complicated, do we need the additional step of installing and using [`pnpm`](https://pnpm.io/)?

#### Current setup

1. Install NodeJS
2. Install `pnpm` globally in Node - using `npm`
3. Use `pnpm` to install dependancies and run dev environment

### Proposal

Remove `pnpm` from the development stack.

#### Onboarding steps

1. Install NodeJS
~~Install `pnpm` globally in Node - using `npm`~~
2. Use `npm` to install dependancies and run dev environment

#### Reasoning

To install `pnpm` requires `npm` which comes as part of the NodeJS installation by default.
Having less tooling and less dependencies is usually a good thing.

We have CI tasks that do not use `pnpm` and require updating to use it.

As `npm` is pre-packaged in Node it is surely the default for most developers.

##### Cons

`pnpm` claims to be 2x faster than `npm`. This probably only matters at a _very_ large scale

Removing `pnpm` will require updates to our documentation. Mostly removals of steps

### Notes

`npm` and `pnpm` are both package managers within the Node ecosystem. Both provide the same functionality.

**Comment by dexslab** [comment]
As of node v16.13 like yarn pnpm is included with node there is one additional step to enable it. 

`corepack enable`

That is all and pnpm is working without additional issues. 
Also look at it like this as well there is a comparison on npm vs pnpm.

All CI has already been converted to use pnpm

[pnpm vs npm](https://pnpm.io/pnpm-vs-npm)

**Reply by dcyoung-dev** [reply]
I may have missed the point 👌

We have a few Github actions that don't use it

- https://github.com/DCC-EX/EX-WebThrottle2/blob/298f2d8eaa55da4ecdaf000d67549b53165403aa/.github/workflows/deploy.yml#L45
- https://github.com/DCC-EX/EX-WebThrottle2/blob/298f2d8eaa55da4ecdaf000d67549b53165403aa/.github/workflows/eslint.yml#L46

Do these need updating?

**Reply by dexslab** [reply]
Yes those do i thought i got everything but i missed those its a quick process to fix it though
